src/data/list/count.lean

```diff
@@ -207,6 +207,20 @@ begin
   { rw [count_cons', count_cons', count_erase_of_ne] }
 end
 
+@[to_additive]
+lemma prod_map_eq_pow_single [monoid β] {l : list α} (a : α)
+  (f : α → β) (hf : ∀ a' ≠ a, a' ∈ l → f a' = 1) :
+  (l.map f).prod = (f a) ^ (l.count a) :=
+begin
```

Comment:
Can you add this special case too?
```lean
@[to_additive]
lemma prod_eq_pow_single [monoid β] {l : list α} (a : α)
  (hf : ∀ a' ≠ a, a' ∈ l → a' = 1) :
  l.prod = a ^ l.count a :=
begin
```
Also, can you add the `multiset` versions of both? It should follow trivially from the list versions.